### ISSUE TITLE
Add & Edit marker

### ISSUE THREAD

**Issue by vmicho** [osmandapp/OsmAnd#16875]
# 🚀 feature request

### Description

Adding a marker with a custom text is currently a long process:
* long press on map
* Marker (this adds a marker with default text on it)
* tap again on the newly created marker
 - if there are multiple selectable items in that region (e.g. another marker, guidepost, currently recording track, other track, route, favourite, POI, ...), select the marker from (sometimes long) list
* Edit
* long press on text/double tap text to select a word/whole text. If the whole text is not selected, tap the popup-menu item 'Select All'
* enter new text
* Save

It becomes extremely annoying after a longer trip with dozens of markers.

### Describe the solution you'd like

Please add a feature Mark & Edit. Ideally to not take any more screen space, this could be for example activated by long pressing the Marker button. Additionally (similar  #16874 ) the whole text could be pre-selected so we can type in a new fresh text.
The above workflow would be simplified to:
* long press on map
* long press on Marker
* type new text
* Save

Would be really appreciated.
Thanks.

**Comment by Zirochkabila** [comment]
@vmicho It seems that the new variant takes as many steps and time as the one that is currently implemented

**Comment by vmicho** [comment]
> @vmicho It seems that the new variant takes as many steps and time as the one that is currently implemented

No way. We talk here 4 VS 8 steps, with the optional step 4 (select marker form list) can be quite annoying. And step 6 takes way too much time as the text is not yet selected and the virt.keyboard is not active.

**Comment by DmitryAlexei** [comment]
Good proposition in #16874 to preselect text when you tap 'edit Map marker'.
Anyway, why do you use map markers instead of offline OSM Notes (https://osmand.net/docs/user/plugins/osm-editing#create--modify-osm-note)?

> > @vmicho It seems that the new variant takes as many steps and time as the one that is currently implemented
> 
> No way. We talk here 4 VS 8 steps, with the optional step 4 (select marker form list) can be quite annoying. And step 6 takes way too much time as the text is not yet selected and the virt.keyboard is not active.

**Comment by vmicho** [comment]
Well, I've played a bit with notes, but it is more suited for online notes. Offline it is more complicated than the current markers :0

**Comment by sonora** [comment]
Have you tried using favorites instead of markers?

A solution could be we abandon the separate logic for markers in general,  instead make the button "Add marker" to result in adding special type/group of favorite where we reduce things to the absolute minimum number of taps possible, which involves pre-filling all  options with a special default.

**Comment by vmicho** [comment]
Well favorites are good for "favorites" (permanent highlight in map), while markers are great for temporary markings in the map.

**Comment by sonora** [comment]
Yes. And that's no _fundamental_ distinction, when you think about it: Just that their nature is more temporary - a kind of favorites which you want to be able to create as quickly as possible  and lose easily when done with them.
